mmv1/products/firestore/api.yaml

```diff
@@ -17,9 +17,120 @@ versions:
   - !ruby/object:Api::Product::Version
     name: ga
     base_url: https://firestore.googleapis.com/v1/
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://firestore.googleapis.com/v1/
 scopes:
   - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Google Cloud Firestore API
+    url: https://console.cloud.google.com/apis/library/firestore.googleapis.com
 objects:
+  - !ruby/object:Api::Resource
+    name: 'Database'
+    min_version: beta
+    base_url: 'projects/{{project}}/databases'
+    create_url: 'projects/{{project}}/databases?databaseId={{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A Cloud Firestore Database. Currently only one database is allowed per
+      cloud project; this database must have a `database_id` of '(default)'.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/firestore/docs/'
+      api: 'https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases'
+    async: !ruby/object:Api::OpAsync
+      actions: ['create','update']
+      operation: !ruby/object:Api::OpAsync::Operation
+        path: 'name'
+        base_url: '{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'response'
+        resource_inside_response: true
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'done'
+        complete: True
```

Comment:
Super nit: I'm seeing different casing for the true/false values in these yaml files. I suspect some of this was copy-and-pasted, and it looks like many of our existing files already have this mixed casing, so up to you if you want to make them consistent.

Reply:
I'm happy to make them consistent. I like all-lowercase for boolean constants, so unless you're aware of a preference for `True` I'll switch all of my new code to use all-lowercase.